src/main/resources/com/mathworks/ci/RunMatlabTestsBuilder/config.jelly

```diff
@@ -2,30 +2,43 @@
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
     
 <f:section title="Generate Test Artifacts">
-  	 <f:entry field="pdfReportChkBx">	
-    	<f:checkbox title="PDF test report " name="pdfReportChkBx" checked="${instance.pdfReportChkBx}"/>	
+    <f:optionalBlock name="pdfReportChkBx" field="pdfReportChkBx" title="PDF test report" checked="${instance.isPdfChecked}">
+  	 <f:entry field="pdfReportFilePath" title="File path: ">	
+    	<f:textbox default="matlabTestArtifacts/testreport.pdf"/>
      </f:entry>
-
-	 <f:entry field="tapChkBx">
-    	<f:checkbox title="TAP test results" name="tapChkBx" checked="${instance.tapChkBx}"/>
+    </f:optionalBlock>
+     
+    <f:optionalBlock name="tapChkBx" field="tapChkBx" title="TAP test results" checked="${instance.isTapChecked}">   	
```

Comment:
I don't see how/why the jelly "queries" the isTapChecked method. It seems like the selections in the UI should set the artifacts, but I don't see why the jelly would need to query them.

Reply:
Usually jelly queries the values to retain the previous selected values ans reflect it on UI when User revisits the configuration page after performing save operation on page. hence to show whether this option was checked or not we had to call this. However i have replaced it with inline expression in jelly that actually checks for the object for null and set the value accordingly.